### ISSUE TITLE
Failed to create TOC

### ISSUE THREAD

**Issue by shahinam** [thlorenz/doctoc#94]
Here is the error:
```
/usr/lib/node_modules/doctoc/node_modules/markdown-to-ast/node_modules/mdast/node_modules/vfile/index.js:537
        throw err;
              ^
46:60: Incorrectly eaten value: please report this warning on http://git.io/vUYWz
```

**Comment by thlorenz** [comment]
Could you please provide the markdown for the file you're trying to doctoc?
Once you do that @jmanuel1 could have a look since he introduced the markdown parsing lib (assuming you have time @jmanuel1 :) ).

**Comment by shahinam** [comment]
Thanks for quick reply. My file is very simple using h1 and h2.
Sorry I am working on a client project, so can not post original file.

```
# Heading One
## Subheading 1
Some text

## Subheading 2
Some text
```

**Comment by cztomczak** [comment]
I have the same error. It fails for these .md files (and lots of more...):
* https://raw.githubusercontent.com/cztomczak/cefpython/62cc77afcea694af151f182ba19d339f8800c0df/docs/objects/Frame.md
* https://raw.githubusercontent.com/cztomczak/cefpython/62cc77afcea694af151f182ba19d339f8800c0df/docs/modules/cefpython.md

Looks like it should be a simple task to parse lines starting with "#", "#, "###", but DocToc fails to do that.

**Comment by jmanuel1** [comment]
Seems to be an error with an older version of `markdown-to-ast` (3.0.7) and its interaction with `woorm/remark` (which was called mdast at the time). I couldn't get the error to happen in master though.

**Comment by azeemba** [comment]
I updated the dependency on `markdown-to-ast` to 3.2.3 and the issue was fixed for me.

If anyone is looking for a workaround, the following should do temporarily:

```
$ npm install -g azeemba/doctoc
```

**Comment by jez** [comment]
@azeemba Do you want to make a pull request with your fix? If updating the dependencies was all it takes we could probably get it reviewed quickly enough.

**Comment by jez** [comment]
Hm, @shahinam the test case you provide does not fail; doctoc is able to parse and transform it correctly. However, the test cases that @cztomczak provides do fail with the error specified.

From those tests I wrote a smaller test readme (~10 lines) that fails. I was able to confirm that upgrading as @azeemba mentioned indeed fixes the issue, and I have a pull request open (see above reference) that tests for and fixes the issue.
